src/cowboy_compress_h.erl

```diff
@@ -72,21 +72,26 @@ early_error(StreamID, Reason, PartialReq, Resp, Opts) ->
 
 %% Check if the client supports decoding of gzip responses.
 check_req(Req) ->
-	%% @todo Probably shouldn't unconditionally crash on failure.
-	case cowboy_req:parse_header(<<"accept-encoding">>, Req) of
-		%% Client doesn't support any compression algorithm.
-		undefined ->
-			#state{compress=undefined};
-		Encodings ->
-			%% We only support gzip so look for it specifically.
-			%% @todo A recipient SHOULD consider "x-gzip" to be
-			%% equivalent to "gzip". (RFC7230 4.2.3)
-			case [E || E={<<"gzip">>, Q} <- Encodings, Q =/= 0] of
-				[] ->
-					#state{compress=undefined};
-				_ ->
-					#state{compress=gzip}
-			end
+	try
+		case cowboy_req:parse_header(<<"accept-encoding">>, Req) of
```

Comment:
Don't wrap the case in a try, replace the case with the try. `try cowboy_req:parse... of ... catch`

Reply:
Nice one! [done here](https://github.com/ninenines/cowboy/compare/deb6dba33003528e934124c5d0827aa5d68eead7..8ffbe073c294743a5ce77ba2d637697b53306dcb).